services/user/Supervisor/ui/src/pluginHost.ts

```diff
@@ -140,7 +140,12 @@ export class PluginHost implements HostInterface {
     ): Result<string, RecoverableErrorPayload> {
         const res = this.postGraphQL(`${this.self.origin}/graphql`, graphql);
         if ("body" in res) {
-            return res.body;
+            if (res.body.includes("errors")) { // works?
+                const json = JSON.parse(res.body);
+                throw this.recoverableError(json.errors.message);
```

Comment:
If you're going to throw instead of return, then you should `throw new Error`. 
`recoverableError` is `return`-ed when the error should be handleable (using `?` or equivalent) within the plugin code.